pyboy/plugins/game_wrapper_kirby_dream_land.py

```diff
@@ -0,0 +1,172 @@
+#
+# License: See LICENSE.md file
+# GitHub: https://github.com/Baekalfen/PyBoy
+#
+__pdoc__ = {
+    "GameWrapperKirbyDreamLand.cartridge_title": False,
+    "GameWrapperKirbyDreamLand.post_tick": False,
+}
+
+import logging
+from math import floor
+from array import array
+
+import numpy as np
+from pyboy.utils import WindowEvent
+
+from .base_plugin import PyBoyGameWrapper
+
+logger = logging.getLogger(__name__)
+
+try:
+    from cython import compiled
+    cythonmode = compiled
+except ImportError:
+    cythonmode = False
+
+
+class GameWrapperKirbyDreamLand(PyBoyGameWrapper):
+    """
+    This class wraps Kirby Dream Land, and provides easy access to score and a "fitness" score for AIs.
+
+    If you call `print` on an instance of this object, it will show an overview of everything this object provides.
+    """
+    cartridge_title = "KIRBY DREAM LA"
+
+    def __init__(self, *args, **kwargs):
+        self.shape = (20, 16)
+        """The shape of the game area"""
+        self.score = 0
+        """The score provided by the game"""
+        self.health = 0
+        """The health provided by the game"""
+        self.lives_left = 0
+        """The lives remaining provided by the game"""
+        self.fitness = 0
+        """
+        A built-in fitness scoring. The scoring is equals to `score`.
+
+        .. math::
+            fitness = score * health * lives_left
+        """
+        super().__init__(*args, game_area_section=(0, 0) + self.shape, game_area_wrap_around=True, **kwargs)
+
+    def post_tick(self):
+        self._tile_cache_invalid = True
+        self._sprite_cache_invalid = True
+
+        multiplier = 100000
+        index = 0
+        score = 0
+        while multiplier > 10:
+          score += self.pyboy.get_memory_value(0xD06F + index) * multiplier
+          multiplier /= 10
+          index += 1
```

Comment:
I can't tell what this part is doing

Reply:
The score is stored between `0xd06F`and five bytes later, with every digit of the score taking one byte (later used for displaying the score on the screen).

So if the score is `007250`, I multiply each digit (`0`, `0`, `7`, `2` etc.) by the multiplier (`100000`, `10000`, `1000`, `100` etc.) to get the score value.

Hope this makes sense - there might be a better way of achieving this - I think you already employ a similar approach when reading the score from the screen and using the tile indexes.

Reply:
Ah, of course! At a glance, I couldn't figure out what was going to be exponential like that.